infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -492,7 +493,13 @@ public void sealServerWithEpoch(long epoch) {
         );
 
         try {
-            batchProcessor.addTask(BatchWriterOperation.Type.SEAL, batchProcessorReq).join();
+            if(batchProcessor.getContext().getStatus() != BatchProcessorStatus.BP_STATUS_ERROR){
+                batchProcessor.addTask(BatchWriterOperation.Type.SEAL, batchProcessorReq).join();
```

Comment:
We might need to explore other approaches for this problem. This looks like race where the status can be set to `BP_STATUS_ERROR` after this check is performed. Then you are still blocked indefinitely on the call to `join()`.

Reply:
We should be able to avoid the race by either synchronizing the `process()` of the BatchProcessor on the context, or alternatively just adding a timeout to the `join()` and handling the exception to move the epoch, avoiding relying on the status all together. Either way is still hacky imo so I am open to different solution.